document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoGroupTypeExpressionParser.java

```diff
@@ -41,10 +46,32 @@ public Map<String, Object> visit(final IdentifierExpression expression) {
     return Map.of(key, PREFIX + identifier);
   }
 
-  public static BasicDBObject getGroupClause(final Query query) {
+  public static List<BasicDBObject> getGroupClauses(final Query query) {
     final List<SelectionSpec> selectionSpecs = query.getSelections();
     final List<GroupTypeExpression> expressions = query.getAggregations();
 
+    final List<BasicDBObject> basicDBObjects = new ArrayList<>();
+
+    final List<SelectionSpec> functionExpressionSelectionWithGroupBys =
+        getFunctionExpressionSelectionWithGroupBys(selectionSpecs, expressions);
+
+    if (!functionExpressionSelectionWithGroupBys.isEmpty()) {
+      MongoSelectTypeExpressionParser parser =
+          new MongoIdentifierPrefixingParser(
+              new MongoIdentifierExpressionParser(new MongoFunctionExpressionParser()));
+      Map<String, Object> addFields =
+          functionExpressionSelectionWithGroupBys.stream()
+              .map(spec -> MongoGroupTypeExpressionParser.parse(parser, spec))
+              .reduce(
+                  new LinkedHashMap<>(),
```

Comment:
Any reason for a linked hashmap? Is the order important here?

Reply:
no reason, i just copy pasted the reduce used below
don't think the order is important